src/shell_minimal.js

```diff
@@ -158,20 +153,20 @@ var ENVIRONMENT_IS_WORKER = typeof importScripts == 'function',
 var _scriptDir = (typeof document != 'undefined') ? document.currentScript?.src : undefined;
 #endif
 
-if (ENVIRONMENT_IS_WORKER) {
-  _scriptDir = self.location.href;
-}
 #if ENVIRONMENT_MAY_BE_NODE
-else if (ENVIRONMENT_IS_NODE) {
+if (ENVIRONMENT_IS_NODE) {
+  var worker_threads = require('worker_threads');
+  global.Worker = worker_threads.Worker;
+  ENVIRONMENT_IS_WORKER = !worker_threads.isMainThread;
+  // Under node we set `workerData` to `em-pthread` to signal that the worker
+  // is hosting a pthread.
+  ENVIRONMENT_IS_PTHREAD = ENVIRONMENT_IS_WORKER && worker_threads['workerData'] == 'em-pthread'
```

Comment:
Sorry, for consistency should this be moved too?
It looks like there's no `createRequire` in shell_minimal.js?

Reply:
Yup, done.

WRT to adding `createRequire` to `shell_minimal.js` I guess that would be separate improvment/extension to `MINIMAL_RUNTIME`, unrelated to this change.